src/i2c/peripheral.rs

```diff
@@ -209,3 +224,62 @@ where
         Ok(())
     }
 }
+
+/// I2C error
+#[derive(Debug, Eq, PartialEq)]
+pub enum Error {
+    /// Event Timeout
+    ///
+    /// Corresponds to the EVENTTIMEOUT flag in the STAT register.
+    EventTimeout,
+
+    /// Master Arbitration Loss
+    ///
+    /// Corresponds to the MSTARBLOSS flag in the STAT register.
+    MasterArbitrationLoss,
+
+    /// Master Start/Stop Error
+    ///
+    /// Corresponds to the MSTSTSTPERR flag in the STAT register.
+    MasterStartStopError,
+
+    /// Monitor Overflow
+    ///
+    /// Corresponds to the MONOV flag in the STAT register.
+    MonitorOverflow,
+
+    /// SCL Timeout
+    ///
+    /// Corresponds to the SCLTIMEOUT flag in the STAT register.
+    SclTimeout,
+}
+
+impl Error {
+    fn read<I: Instance>(i2c: &I) -> Option<Self> {
+        let stat = i2c.stat.read();
+
+        // Check for error flags. If one is set, clear it and return the error.
+        if stat.mstarbloss().bit_is_set() {
+            i2c.stat.write(|w| w.mstarbloss().set_bit());
+            return Some(Self::MasterArbitrationLoss);
+        }
+        if stat.mstststperr().bit_is_set() {
+            i2c.stat.write(|w| w.mstststperr().set_bit());
+            return Some(Self::MasterStartStopError);
+        }
+        if stat.monov().bit_is_set() {
```

Comment:
Not sure if it makes sense to include this, since it's only relevant in monitor mode and we don't check the slave flags as well. (not particularly important)

Reply:
I figured it's better to check a useless flag now, then to forget to add the check later. I thought I had added checks for all the slave error conditions as well, but I'll double-check later.

I don't feel strongly about this, but since you don't seem to feel strongly either, I'm inclined to leave the checks in.

Reply:
Just checked, all slave error conditions are handled. Probably misread some of the slave non-error conditions.